src/wallet/wallet.cpp

```diff
@@ -1224,7 +1224,14 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate)
             ReadBlockFromDisk(block, pindex, Params().GetConsensus());
             BOOST_FOREACH(CTransaction& tx, block.vtx)
```

Comment:
Right. This loop is stupid at this point. The transaction always have a `posInBlock`. Will update.